### PR TITLE
HELM-348: Create documentation for each of the panels provided by Helm

### PR DESCRIPTION
**Pull request by Bonrob2**
# Pull Request Check Sheet

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new feature, is there documentation?
* [ ] If this is a new feature or substantial change, are there tests that cover it?

# Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

Once the reviewer(s) accept the PR and the branch passes continuous integration in Bamboo, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

# External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/HELM-348
* Continuous Integration: [CircleCI](https://circleci.com/gh/OpenNMS/opennms-helm)

### CODE REVIEW

**Comment by Bonrob2** [comment]
@emarsh1995 and @mmahacek note that I updated some of the formatting to more towards our new style guidelines, but will do the rest as part of an overhaul of the Helm docs for the next major Helm release.

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
+* xref:bw-ds-setup[Set up datasources].
+* xref:bw-dash-setup[Create a dashboard].
+* xref:bw-panel-create[Create a panel].
 
-
-The walk-through assumes that you have:
+The walk-through assumes that you have the following:
 
 * An instance of Grafana with the {page-component-title} application installed.
 * An instance of OpenNMS that is reachable from the Grafana instance above.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:17]:
```suggestion
* An instance of OpenNMS that is reachable from your Grafana instance.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 2]
```diff
@@ -23,7 +22,7 @@ If you already use the OpenNMS Data Source plugin with Grafana, see xref:migrati
 [[bw-app-setup]]
 == Set up plugin
 
-In this section, you will enable the {page-component-title} plugin in Grafana and create <<custom actions>> to help with alarm diagnosis.
+In this section, you will enable the {page-component-title} plugin in Grafana and create <<custom-actions, custom actions>> to help with alarm diagnosis.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:25]:
```suggestion
In this section, you will enable the {page-component-title} plugin in Grafana and create <<custom-actions, custom actions>> to help with alarm diagnosis:
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 3]
```diff
@@ -39,9 +38,10 @@ If the issue persists, make sure the plugin is in the grafana plugins directory
 . In the *Config* tab, click *Enable*.
 +
 A *Disable* button indicates the plugin is already enabled.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:40]:
```suggestion
A *Disable* button indicates that the plugin is already enabled.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 4]
```diff
@@ -39,9 +38,10 @@ If the issue persists, make sure the plugin is in the grafana plugins directory
...
-Skip to the next step (create <<custom actions>>).
+Skip to the next step (create <<custom-actions, custom actions>>).
 You may also choose *Update* for the latest version of {page-component-title}.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:42]: Why do we tell the user to skip to the next step here? This seems to be the last step in this part of the procedure, unless them choosing to update the plugin should also be a step?

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 5]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
 For information on displaying additional or different types of data, see xref:bw-panel-create[creating a panel].
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:118]:
```suggestion
For information on displaying other data in a dashboard, see xref:bw-panel-create[creating a panel].
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 6]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 
 . Create a new dashboard by clicking the *+* sign in the left sidebar to open the Create menu and choose *Dashboard*.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:120]:
```suggestion
. In the left menu, click the *+* symbol and choose *Dashboard* to create a new dashboard.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 7]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
-. Click *Add New Panel* and configure the panel in the lower half of the screen:
+. Click *Add Panel>Add an empty panel* and configure the panel in the lower half of the screen:
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:121]:
```suggestion
. Click *Add Panel > Add an Empty Panel* and configure its settings in the lower half of the screen:
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 8]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 +
 image::gf-panel-config.png[Configure Panel, 500]
 
 . In the *Query* drop-down, select the datasource we previously created in xref:bw-ds-setup[Datasource Setup].
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:125]:
```suggestion
. Select the datasource that you created in xref:bw-ds-setup[Datasource Setup] from the *Query* list.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 9]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 . Select *Alarms* in the Select list.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:126]:
```suggestion
. Select *Alarms* from the *Select* list.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 10]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 .. Configure additional information and add filters as desired (click *+Query* to add queries).
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:127]:
```suggestion
** Configure additional information and add filters as desired (click *+ Query* to add queries).
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 11]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 
-. In the *Panel* tab, add a name for the panel under menu:Settings[Panel].
+. In the *Panel Options* area on the right, add a name for the panel in the *Title* box.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:129]:
```suggestion
. Under Panel Options, type the panel's name in the *Title* box.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 12]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 . Under *Visualization*, select *Alarm Table*.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:130]:
```suggestion
. Under Visualization, select *Alarm Table*.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 13]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 The graph appears in the top of the panel.
-.. You can further customize the graph by specifying information in the *Panel* tab.
+.. You can further customize the graph by specifying information in the *Panel Options* area.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:132]:
```suggestion
** You can further customize the graph by specifying information under Panel Options.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 14]
```diff
@@ -145,9 +145,20 @@ Create more panels to display other data, or xref:importing.adoc#gs-dashboard-im
 == Create a panel
 
 A panel is the part of your dashboard that displays the fault and performance management data you specify.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:147]:
```suggestion
A panel is the part of your dashboard that displays the fault and performance management data that you specify.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 15]
```diff
@@ -145,9 +145,20 @@ Create more panels to display other data, or xref:importing.adoc#gs-dashboard-im
...
+
 Before you add a panel, determine the type of data you want to display (performance, metrics, alarms/nodes), and make sure that you have xref:bw-ds-setup[set up the associated datasource].
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:149]:
```suggestion
Before you add a panel, determine the type of data that you want to display (performance, metrics, alarms or nodes, and so on) and make sure that you have xref:bw-ds-setup[set up the associated datasource].
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 16]
```diff
@@ -145,9 +145,20 @@ Create more panels to display other data, or xref:importing.adoc#gs-dashboard-im
...
+
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:150]:
```suggestion
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 17]
```diff
@@ -145,9 +145,20 @@ Create more panels to display other data, or xref:importing.adoc#gs-dashboard-im
...
 Default graph types in Grafana work only with the appropriate datasource (for example, an alarm table visualization will not display data if a flow datasource is specified).
 
+{page-component-title} includes the following custom panels:
+
+* Alarm Histogram
+* Alarm Table
+* Filter Panel
+* Flow Histogram
+
+For more information about them, see xref:panel_configuration:index.adoc#custom-panel[Default custom panels].
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:160]:
```suggestion
For more information about custom panel types, see xref:panel_configuration:index.adoc#custom-panel[Default custom panels].
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 18]
```diff
@@ -145,9 +145,20 @@ Create more panels to display other data, or xref:importing.adoc#gs-dashboard-im
...
+
 . In your dashboard, click the *Add Panel* icon and select *Add New Panel*.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:162]:
```suggestion
Follow the steps below to create a new panel in your existing dashboard:

. Navigate to your dashboard and click the *Add Panel* symbol.
. Click *Add New Panel*.
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 19]
```diff
@@ -12,6 +12,31 @@ The available configuration options depend on the type of datasource associated
...
-* xref:dynamic-dashboard.adoc[dynamic dashboards]
-* xref:expressions.adoc[working with expressions]
+* xref:forecasting.adoc[Forecast metrics]
+* xref:dynamic-dashboard.adoc[Dynamic dashboards]
+* xref:expressions.adoc[Working with expressions]
+
+[[custom-panel]]
+== Default custom panels
+
+{page-component-title} comes with several pre-configured, custom panels, described in this section.
```
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:22]:
```suggestion
{page-component-title} comes with several pre-configured, custom panels.
They are described in this section.
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 20]
```diff
@@ -12,6 +12,31 @@ The available configuration options depend on the type of datasource associated
...
+
+=== Alarm Histogram
+Bar chart that shows the distribution of alarms by severity level.
+
+image::alarm-histogram.png[Horizontal bar graph showing distribution of critical and major alarms, 500]
```
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:27]:
```suggestion
Alarm Histogram:: Bar chart that shows the distribution of alarms by severity level.
+
image::alarm-histogram.png[Horizontal bar graph showing distribution of critical and major alarms, 500]
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 21]
```diff
@@ -12,6 +12,31 @@ The available configuration options depend on the type of datasource associated
...
+
+=== Alarm Table
+Displays alarms in table format.
+
+image::alarm-panel.png[Table of unacknowledged alarms and a table of acknowledged alamrs, 500]
```
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:32]:
```suggestion
Alarm Table:: Displays alarms in table format.
+
image::alarm-panel.png[Two tables: one displays unacknowledged alarms, and the other displays acknowledged alarms, 500]
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 22]
```diff
@@ -12,6 +12,31 @@ The available configuration options depend on the type of datasource associated
...
+
+=== Filter Panel
+A filter panel lets you specify data to include other panels.
+
+image::filter-panel.png[Blank fields in a filter panel, 500]
```
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:37]:
```suggestion
Filter Panel:: A filter panel lets you specify data to include other panels.
+
image::filter-panel.png[Blank fields in a filter panel, 500]
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 23]
```diff
@@ -12,6 +12,31 @@ The available configuration options depend on the type of datasource associated
...
+
+=== Flow Histogram
+Bar chart that shows the distribution of flows data.
+
+image::flow-histogram.png[Horizontal bar graph that displays network flow data, 500]
```
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:42]:
```suggestion
Flow Histogram:: Bar chart that shows the distribution of flows data.
+
image::flow-histogram.png[Horizontal bar graph that displays network flow data, 500]
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 24]
```diff
@@ -118,18 +118,18 @@ This section describes how to create a dashboard with an alarms table.
...
 
-. In the *Panel* tab, add a name for the panel under menu:Settings[Panel].
+. In the *Panel Options* area on the right, add a name for the panel in the *Title* box.
 . Under *Visualization*, select *Alarm Table*.
 The graph appears in the top of the panel.
-.. You can further customize the graph by specifying information in the *Panel* tab.
+.. You can further customize the graph by specifying information in the *Panel Options* area.
 
 . In the top-right menu, click *Save*.
 . Enter a name for the dashboard and click *Save*.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:135]:
```suggestion
. Type a name for the dashboard and click *Save*.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 25]
```diff
@@ -39,9 +38,10 @@ If the issue persists, make sure the plugin is in the grafana plugins directory
...
 +
 A *Disable* button indicates the plugin is already enabled.
-Skip to the next step (create <<custom actions>>).
+Skip to the next step (create <<custom-actions, custom actions>>).
 You may also choose *Update* for the latest version of {page-component-title}.
 
+[[custom-actions]]
 === Custom actions
 
 A custom action is a user-defined operation, such as a link to a node detail page or a Wiki knowledge base article, that might help users better diagnose an alarm.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:47]:
```suggestion
A custom action is a user-defined operation, such as a link to a node detail page or a knowledge base article, that might help users better diagnose an alarm.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 26]
```diff
@@ -23,7 +22,7 @@ If you already use the OpenNMS Data Source plugin with Grafana, see xref:migrati
 [[bw-app-setup]]
 == Set up plugin
 
-In this section, you will enable the {page-component-title} plugin in Grafana and create <<custom actions>> to help with alarm diagnosis.
+In this section, you will enable the {page-component-title} plugin in Grafana and create <<custom-actions, custom actions>> to help with alarm diagnosis.
 
 . Navigate to the home page of your Grafana instance.
 . In the left menu, click the *Configuration* icon, and click *Plugins*:
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:28]:
```suggestion
. In the left menu, click the *Configuration* symbol, and click *Plugins*:
```

**Review by emarsh1995 — CHANGES_REQUESTED** [review]

File: docs/modules/getting_started/pages/basic_walkthrough.adoc (hunk 4 above)
Comment (Bonrob2) [reply docs/modules/getting_started/pages/basic_walkthrough.adoc:42]: Good catch. I do not know. Updated.

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 27]
```diff
@@ -38,13 +37,13 @@ If the issue persists, make sure the plugin is in the grafana plugins directory
 
 . In the *Config* tab, click *Enable*.
 +
-A *Disable* button indicates the plugin is already enabled.
-Skip to the next step (create <<custom actions>>).
+A *Disable* button indicates that the plugin is already enabled.
 You may also choose *Update* for the latest version of {page-component-title}.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:41]: Does this update OpenNMS Horizon/Meridian itself, or the plugin that the user is setting up?

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 28]
```diff
@@ -115,24 +114,24 @@ If the message indicates a failure, review the information you entered and click
...
+. Select *Alarms* from the *Select* list.
+** Configure additional information and add filters as desired (click *+ Query* to add queries).
 
-. In the *Panel* tab, add a name for the panel under menu:Settings[Panel].
-. Under *Visualization*, select *Alarm Table*.
+. Under Panel Options, type the panel's name in the *Title* box.
+. Under Visualization, select *Alarm Table*.
 The graph appears in the top of the panel.
-.. You can further customize the graph by specifying information in the *Panel* tab.
+** You can further customize the graph by specifying information under Panel Options.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:132]:
```suggestion
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 29]
```diff
@@ -144,11 +143,24 @@ Create more panels to display other data, or xref:importing.adoc#gs-dashboard-im
...
+* Filter Panel
+* Flow Histogram
+
+For more information about custom panel types, see xref:panel_configuration:index.adoc#custom-panel[Default custom panels].
+
+Follow the steps below to create a new panel in your existing dashboard:
+
+. Navigate to your dashboard and click the *Add Panel* symbol.
+. Click *Add New Panel*.
 . Select the datasource from the *Query* list.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:165]:
```suggestion
```

**Review by emarsh1995 — CHANGES_REQUESTED** [review]
A few pedantic nits and a question. Thanks! :)

File: docs/modules/getting_started/pages/basic_walkthrough.adoc (hunk 27 above)
Comment (Bonrob2) [reply docs/modules/getting_started/pages/basic_walkthrough.adoc:41]: Good point. Just the plugin.

Comment (Bonrob2) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:41]:
```suggestion
You may also choose *Update* for the latest version of the {page-component-title} plugin.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 30]
```diff
@@ -144,13 +142,25 @@ Create more panels to display other data, or xref:importing.adoc#gs-dashboard-im
...
+* Filter Panel
+* Flow Histogram
+
+For more information about custom panel types, see xref:panel_configuration:index.adoc#custom-panel[Default custom panels].
 
+Follow the steps below to create a new panel in your existing dashboard:
+
+. Navigate to your dashboard and click the *Add Panel* symbol.
+. Click *Add New Panel*.
+. Select the datasource from the *Query* list.
```
Comment (emarsh1995) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:163]:
```suggestion
. Select a datasource from the *Query* list.
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 31]
```diff
@@ -12,6 +12,28 @@ The available configuration options depend on the type of datasource associated
 
 This section describes the following types of panel configuration:
 
-* xref:forecasting.adoc[forecast metrics]
-* xref:dynamic-dashboard.adoc[dynamic dashboards]
-* xref:expressions.adoc[working with expressions]
+* xref:forecasting.adoc[Forecast metrics]
+* xref:dynamic-dashboard.adoc[Dynamic dashboards]
+* xref:expressions.adoc[Working with expressions]
```
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:17]:
```suggestion
* xref:forecasting.adoc[Forecast metrics].
* xref:dynamic-dashboard.adoc[Dynamic dashboards].
* xref:expressions.adoc[Working with expressions].
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 32]
```diff
@@ -12,6 +12,28 @@ The available configuration options depend on the type of datasource associated
...
+
+Alarm Histogram:: Bar chart that shows the distribution of alarms by severity level.
++
+image::alarm-histogram.png[Horizontal bar graph showing distribution of critical and major alarms, 500]
+
+Alarm Table:: Displays alarms in table format.
++
+image::alarm-panel.png[Two tables: one displays unacknowledged alarms, and the other displays acknowledged alarms, 500]
+
+Filter Panel:: A filter panel lets you specify data to include other panels.
```
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:33]:
```suggestion
Filter Panel:: Panel that lets you specify data to include other panels.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 33]
```diff
@@ -4,29 +4,29 @@
...
+* xref:bw-app-setup[Set up the application and custom actions].
+* xref:bw-ds-setup[Set up datasources].
+* xref:bw-dash-setup[Create a dashboard].
+* xref:bw-panel-create[Create a panel].
 
-
-The walk-through assumes that you have:
+This walk-through assumes that you have the following:
 
 * An instance of Grafana with the {page-component-title} application installed.
```
Comment (mmahacek) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:17]:
```suggestion
* An instance of Grafana with the {page-component-title} plugin installed.
```

File: docs/modules/panel_configuration/pages/index.adoc [hunk 34]
```diff
@@ -12,6 +12,28 @@ The available configuration options depend on the type of datasource associated
...
+
+Alarm Table:: Displays alarms in table format.
++
+image::alarm-panel.png[Two tables: one displays unacknowledged alarms, and the other displays acknowledged alarms, 500]
+
+Filter Panel:: Panel that lets you specify data to include in other panels.
++
+image::filter-panel.png[Blank fields in a filter panel, 500]
+
+Flow Histogram:: Bar chart that shows the distribution of flows data.
```
Comment (mmahacek) [inline docs/modules/panel_configuration/pages/index.adoc:37]:
```suggestion
Flow Histogram:: Bar chart that shows the distribution of flow data.
This panel expects the Flow Datasource query to have a `asTableSummary()` transform included.
```

**Review by emarsh1995 — APPROVED** [review]
LGTM! Thank you!

**Review by mmahacek — APPROVED** [review]

**Pull request merged**